redis/connection.py

```diff
@@ -2146,66 +2346,173 @@ def release(self, connection):
-                    self._disconnect_and_update_connection_for_reconnect(
-                        conn, tmp_host_address, tmp_relax_timeout
-                    )
+        When this method is called the pool will already be locked, so getting the pool lock inside is not needed.
 
-    def update_connections_current_timeout(self, relax_timeout: Optional[float] = None):
-        logging.debug(
-            f"***** Blocking Pool --> Updating timeouts. relax_timeout: {relax_timeout}"
-        )
+        :param tmp_host_address: The temporary host address to use for the connection.
+        :param tmp_relax_timeout: The relax timeout to use for the connection.
+        """
+        existing_connections = self.pool.queue
 
-        with self._lock:
-            for conn in tuple(self._connections):
-                self._update_connection_timeout(conn, relax_timeout)
+        for conn in existing_connections:
+            if conn:
+                if moving_address_src and conn.getpeername() != moving_address_src:
+                    continue
+                self._disconnect_and_update_connection_for_reconnect(
+                    conn, tmp_host_address, tmp_relax_timeout
+                )
 
-    def update_connections_tmp_settings(
+    def update_connections_current_timeout(
         self,
-        tmp_host_address: Optional[str] = None,
-        tmp_relax_timeout: Optional[float] = None,
+        relax_timeout: Optional[float] = None,
+        matching_address: Optional[str] = None,
+        address_type_to_match: Literal["connected", "configured"] = "connected",
+        include_free_connections: bool = False,
     ):
-        with self._lock:
+        """
+        Update the timeout for the current socket.
+        This is used when a cluster node is migrated to a different address.
+
+        When this method is called the pool will already be locked, so getting the pool lock inside is not needed.
+
+        :param relax_timeout: The relax timeout to use for the connection.
+        :param include_free_connections: Whether to include available connections in the update.
+        """
+        if include_free_connections:
             for conn in tuple(self._connections):
-                self._update_connection_tmp_settings(
-                    conn, tmp_host_address, tmp_relax_timeout
-                )
+                if address_type_to_match == "connected":
+                    if matching_address and conn.getpeername() != matching_address:
+                        continue
+                else:
+                    if matching_address and conn.host != matching_address:
+                        continue
+                conn.update_current_socket_timeout(relax_timeout)
+        else:
+            connections_in_queue = {conn for conn in self.pool.queue if conn}
+            for conn in self._connections:
+                if conn not in connections_in_queue:
+                    if address_type_to_match == "connected":
```

Comment:
There's a lot of duplicate of this part:

```
                    if address_type_to_match == "connected":
                        if matching_address and conn.getpeername() != matching_address:
                            continue
```

Maybe it makes to create a method that filter connections by this criteria and returns filtered?